### PR TITLE
Export/import of a single repository (Content Management)

### PR DESCRIPTION
**Pull request by Lennonka**
Cherry-pick into:

* [x] Foreman 3.2
* [x] Foreman 3.1
* There will be one more PR for 2.5

### CODE REVIEW

**Comment by maximiliankolb** [comment]
original PR: https://github.com/theforeman/foreman-documentation/pull/1143

**Comment by Lennonka** [comment]
@parthaa Can you please review? If we want this published next week, we need to merge it soonish :)

File: guides/common/modules/proc_exporting-a-repository.adoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* Two JSON files containing repository metadata.
+* An archive file containing the contents of the repository from the Library environment of the organization.
+
+You need all the files, `tar.gz`, `toc.json` and `metadata.json`, to be able to import.
+
+.Prerequisites
+
+To export the contents of a repository, ensure that {ProjectServer} from which you want to export, meets the following conditions:
+
+* Ensure that the export directory has free storage space to accommodate the export.
```
Comment (chris1984) [inline guides/common/modules/proc_exporting-a-repository.adoc:23]: `has enough free storage space`

File: guides/common/modules/proc_exporting-a-repository.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+* Ensure that the `/var/lib/pulp/exports` directory has free storage space equivalent to the size of the repositories being exported for temporary files created during the export process.
```
Comment (chris1984) [inline guides/common/modules/proc_exporting-a-repository.adoc:24]: has enough free storage space

File: guides/common/modules/con_iss-import-export-cheat-sheet.adoc [hunk 3]
```diff
@@ -9,6 +9,10 @@
 |Fully Export a Content View version | `hammer content-export complete version --content-view="_My_Content_View_" --version=1.0 --organization="_My_Organization_"`
 |Incrementally Export a Content View version (assuming you have already exported something)| `hammer content-export incremental version --content-view="_My_Content_View_" --version=2.0 --organization="_My_Organization_"`
 
+|Fully Export a Repository| `hammer content-export complete library --product="_My_Product_" --name="_My_Repository" --organization="_My_Organization_"`
```
Comment (rmynar) [inline guides/common/modules/con_iss-import-export-cheat-sheet.adoc:12]: To export complete repository
`hammer content-export complete repository ...`
should be used.

File: guides/common/modules/con_iss-import-export-cheat-sheet.adoc [hunk 4]
```diff
@@ -9,6 +9,10 @@
...
+
+|Incrementally Export a Repository (assuming you have already exported something)|`hammer content-export incremental --product="_My_Product_" --name="_My_Repository" --organization="_My_Organization_"`
```
Comment (rmynar) [inline guides/common/modules/con_iss-import-export-cheat-sheet.adoc:14]: What to export is missing here.
`hammer content-export incremental repository ...`

File: guides/common/modules/proc_exporting-a-repository-incrementally.adoc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+Incremental exports typically result in smaller archive files than the full exports.
+
+The example below shows incremental export of a repository in the Library lifecycle environment.
+
+.Procedure
+. Create an incremental export:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export incremental repository\
```
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/common/modules/proc_exporting-a-repository-incrementally.adoc:17]: Definitely wasn't, thanks for catching that :D

File: guides/common/modules/proc_exporting-a-repository.adoc [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+total 68M
+-rw-r--r--. 1 pulp pulp 68M Mar  2 03:35 export-1e25417c-6d09-49d4-b9a5-23df4db3d52a-20210302_0335.tar.gz
+-rw-r--r--. 1 pulp pulp 333 Mar  2 03:35 export-1e25417c-6d09-49d4-b9a5-23df4db3d52a-20210302_0335-toc.json
+-rw-r--r--. 1 root root 443 Mar  2 03:35 metadata.json
+----
+
+.Export a Repository with Chunking
+
+In many cases the exported content archive may be several gigabytes in size.
+If you want to split it into chunks of smaller size, you can use the `--chunk-size-gb` argument in the export command and limit the size by a value in gigabytes.
```
[the remark this answers is not in the document]
Comment (chris1984) [reply guides/common/modules/proc_exporting-a-repository.adoc:55]: @Lennonka it looks to be a integer, https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_exports_controller.rb#L49

**Review by parthaa — APPROVED** [review]
lgtm

File: guides/common/modules/proc_importing-a-repository.adoc [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+For more information about exporting content of a repository, see xref:Exporting_a_Repository_{context}[].
+
+.Prerequisites
+
+To import a repository to an Organization's library lifecycle environment ensure that {ProjectServer} on which you want to import, meets the following conditions:
+
+* The export files must be in a directory under `/var/lib/pulp/imports`.
+* The directory must have `pulp:pulp` ownership so that Pulp can read and write the `.json` files in that directory.
+* If the export contains any Red Hat repositories, the manifest of the importing organization must contain subscriptions for the products contained within the export.
+* The user importing the content must have the 'Content Importer' Role.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_importing-a-repository.adoc:14]:
```suggestion
* The user importing the content must have the _Content Importer_ Role.
```

Maybe either italic or code? I don't think we currently use quotes to highlight text.

File: guides/common/modules/proc_importing-a-repository.adoc [hunk 8]
```diff
@@ -0,0 +1,45 @@
+[id="Importing_a_Repository_{context}"]
+= Importing a Repository
```
Comment (maximiliankolb) [inline guides/common/modules/proc_importing-a-repository.adoc:2]: Please rename to "Importing a Yum Repository", including the anchor and file name.

File: guides/common/modules/proc_exporting-a-repository.adoc [hunk 9]
```diff
@@ -0,0 +1,74 @@
+[id="Exporting_a_Repository_{context}"]
+= Exporting a Repository
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository.adoc:2]: Please rename to "Exporting a Yum Repository"; including file name and anchor.

File: guides/common/modules/proc_importing-a-repository.adoc [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+. To import the Library content to {ProjectServer}, enter the following command:
++
+[subs="+quotes"]
+----
+# hammer content-import repository \
+--organization="_My_Organization_" \
+--path=/var/lib/pulp/imports/_2021-03-02T03-35-24-00-00_
+----
++
+Note that you must enter the full path `/var/lib/pulp/imports/<path>`.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_importing-a-repository.adoc:43]:
```suggestion
Note that you must enter the full path `/var/lib/pulp/imports/_Your_Exported_Repository_`.
```

File: guides/common/modules/ref_iss-import-export-cheat-sheet.adoc [hunk 11]
```diff
@@ -26,7 +30,7 @@
 |=========================================================
 |Intent | Command
 
-|Import to a Content View version | `hammer content-import version --organization="_My_Organization_" --path=’/var/lib/pulp/imports/dump_dir’`
-
-|Import to an Organization's Library| `hammer content-import library --organization="_My_Organization_" --path=’/var/lib/pulp/imports/dump_dir’`
+|Import to a Content View version | `hammer content-import version --organization="_My_Organization_" --path="/var/lib/pulp/imports/_dump_dir_"`
+|Import a Repository| `hammer content-import repository --organization="_My_Organization_" --path="/var/lib/pulp/imports/_dump_dir_"`
+|Import to an Organization's Library| `hammer content-import library --organization="_My_Organization_" --path="/var/lib/pulp/imports/_dump_dir_"`
```
Comment (maximiliankolb) [inline guides/common/modules/ref_iss-import-export-cheat-sheet.adoc:35]: Maybe also find a nice word than "dump_dir".

**Review by maximiliankolb — COMMENTED** [review]
Some suggestions & I would like to see a successful Github pipeline. I don't know why it didn't start automatically. 🤷

File: guides/common/modules/proc_importing-a-repository.adoc (hunk 8 above)
Comment (Lennonka) [reply guides/common/modules/proc_importing-a-repository.adoc:2]: @maximiliankolb Does "Yum Repository" embrace all the exportable repo types, i.e. RPM, Kickstart and file repositories?

File: guides/common/modules/ref_iss-import-export-cheat-sheet.adoc (hunk 11 above)
Comment (Lennonka) [reply guides/common/modules/ref_iss-import-export-cheat-sheet.adoc]: @maximiliankolb Pls see update and let me know if it works.

File: guides/common/modules/proc_importing-a-repository.adoc (hunk 8 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_importing-a-repository.adoc:2]: OK. If we keep it generic, then please also ensure that it covers DEB content :) See line 4 and line 7. Please ignore the comment for "exporting-a-repository.adoc" but also ensure if covers DEB content.

Comment (Lennonka) [reply guides/common/modules/proc_importing-a-repository.adoc:2]: Got it. Good point!

File: guides/common/modules/ref_iss-import-export-cheat-sheet.adoc (hunk 11 above)
Comment (maximiliankolb) [reply guides/common/modules/ref_iss-import-export-cheat-sheet.adoc]: Perfect.

File: guides/common/modules/proc_exporting-a-repository.adoc [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+* RPM repositories
+* Kickstart repositories
+* Ansible repositories
+* file repositories
+
+You cannot export Docker content
+ifndef::satellite[]
+nor DEB repositories
+endif::[]
+from {ProjectServer}.
```
Comment (Lennonka) [inline guides/common/modules/proc_exporting-a-repository.adoc:18]: @maximiliankolb The situation is that DEB repos are not supported at all. We have verified this with Pulp developers.

Comment (maximiliankolb) [reply guides/common/modules/proc_exporting-a-repository.adoc:18]: Ok, thanks for checking!

**Comment by Lennonka** [comment]
So, if we are all happy, can we merge it perhaps? :D

**Review by maximiliankolb — APPROVED** [review]

**Pull request merged**